webapp/channels/src/components/menu/menu_item.tsx

```diff
@@ -107,7 +116,7 @@ export function MenuItem(props: Props) {
         isLabelsRowLayout,
         children,
         onClick,
-        ...restProps
+        ...otherProps
     } = props;
```

Comment:
(Out of the scope)
If we are doing this just at the beginning of the component, we could directly do it on the arguments of the function.

(i.e. `export function MenuItem({leadingElement, ...}: Props) {`)

Reply:
I'm 0/5 on how we do that myself. Personally, I actually prefer just keeping `props` and referencing things like `props.children` in more places since I think knowing something is a prop is valuable, but I know I'm in the minority there, especially since it can cause issues with hook memoization or at least its ESLint rule

Reply:
I see your point, but in that case, let's not expand it. I am biased towards expanding, because allows you to track a bit better unused props. But that being said... what I am proposing is either... expand it in the function arguments, or don't expand it at all :P